test/functional/feature_signet.py

```diff
@@ -80,20 +88,22 @@ def check_getmininginfo(node_idx, signet_idx):
             assert 'currentblockweight' not in mining_info
             assert_equal(mining_info['networkhashps'], Decimal('0'))
             assert_equal(mining_info['pooledtx'], 0)
-            assert_equal(mining_info['signet_challenge'], self.signets[signet_idx].challenge)
+            assert_equal(mining_info['signet_challenge'], self.signets[signet_idx].internal_challenge)
         check_getmininginfo(node_idx=0, signet_idx=0)
         check_getmininginfo(node_idx=3, signet_idx=1)
         check_getmininginfo(node_idx=4, signet_idx=2)
+        check_getmininginfo(node_idx=7, signet_idx=3)
 
         self.generate(self.nodes[0], 1, sync_fun=self.no_op)
 
         self.log.info("pregenerated signet blocks check")
 
-        height = 0
-        for block in signet_blocks:
-            assert_equal(self.nodes[2].submitblock(block), None)
-            height += 1
-            assert_equal(self.nodes[2].getblockcount(), height)
+        for node_idx in [2, 6]:
```

Comment:
Why change this? Why not also include nodes 0 & 1?

Reply:
I verify that nodes accept blocks mined using the default signet challenge. This test includes one default signet node (2) and another node (6) configured with an extended signet challenge that uses the actual script OP_TRUE (accepts all blocks).

I added a comment with this explanation to the code.